pkg/arvo/sys/vane/gall.hoon

```diff
@@ -53,16 +53,17 @@
 ::
 +$  yoke
   $:  control-duct=duct
+      nonce=@t
```

Comment:
Are we committing to using nonces for app scrys now?  I wasn't sure we'd resolved that discussion.  I'm slightly concerned that running a hash function on every app invocation will be slow, but maybe that's not worth worrying about.

I'd be inclined to push this in one of two directions.  Either:
+ defer all nonce and scry work until later, or
+ go all the way and redo `$bowl` now to include the nonce, so that apps don't need to do an upgrade later.

Reply:
This is a different nonce.  The hash is run only once when starting an app and put in the duct used for all `%pass` moves from that app.  An example of the bug we're avoiding is:

- Start app `%foo`, which starts a timer for ~d1
- Delete app `%foo`
- Less than ~d1 later, start another copy of `%foo`
- `%foo` gets a timer response that it didn't expect because the timer was started by an earlier iteration of the app.

Several other kernel interfaces could cause this, like Clay subscriptions.  Until we make those legible to Gall, we can't cancel them automatically.